src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -2413,6 +2423,8 @@ public override void WriteFieldData(ref ObjectDataBuilder builder, NodeFactory f
             {
                 builder.EmitPointerReloc(factory.SerializedMaximallyConstructableRuntimeTypeObject(TypeRepresented));
             }
+
+            public override bool TargetSupportsWritingFieldData => Type.Context.Target.SupportsRelativePointers;
```

Comment:
public override bool TargetSupportsWritingFieldData => EETypeNode.SupportsFrozenRuntimeTypeInstances(Type.Context.Target);

Reply:
Do you want me to upstream these changes?